src/overlay/Peer.cpp

```diff
@@ -594,19 +593,19 @@ Peer::recvMessage(StellarMessage const& stellarMsg)
         fmt::format("Error RecvMessage T:{} cat:{} {} @{}", stellarMsg.type(),
                     cat, toString(), mApp.getConfig().PEER_PORT);
 
-    mApp.postOnMainThread(
-        [ err, weak, sm = StellarMessage(stellarMsg) ]() {
-            auto self = weak.lock();
-            if (self)
-            {
-                self->recvRawMessage(sm);
-            }
-            else
-            {
-                CLOG(TRACE, "Overlay") << err;
-            }
-        },
-        {catType, fmt::format("{}-{} recvMessage", cat, toString())});
+    mApp.postOnMainThread([ err, weak, sm = StellarMessage(stellarMsg) ]() {
+        auto self = weak.lock();
+        if (self)
+        {
+            self->recvRawMessage(sm);
+        }
+        else
+        {
+            CLOG(TRACE, "Overlay") << err;
+        }
+    },
+                          fmt::format("{}-{} recvMessage", cat, toString()),
```

Comment:
Semi-related to my question on SCP in the scheduler commit.
I *think* that by having a queue per connection for everything (use `toString()`) we're semi-mitigating the issue I raised that some SCP messages will take a very long time.

With the current implementation, the connection that sends the message that allows the node to externalize will get the blame for it, and the node will have to rely on other connections to reach consensus for the next ledger (to a certain extent).

While that seems like a feature, this "stop and go" pattern doesn't seem to be a desirable as it's going to cause weird stalls in other systems that are not aware that certain connections are getting throttled (like the code that asks peers one by one for data).

I think preserving the queue per connection is the right thing, but we should not trigger stalls when externalizing.

Reply:
Outdated comment, resolving